pkgs/by-name/ro/rockbox-utility/package.nix

```
@@ -1,74 +1,56 @@
-{ lib
-, stdenv
-, fetchurl
-, cryptopp
-, libusb1
-, makeWrapper
-, pkg-config
-, qt5
-, withEspeak ? false, espeak ? null
+{
+  lib,
+  stdenv,
+  fetchgit,
+  cmake,
+  pkg-config,
+  qt5,
+  makeWrapper,
+  cryptopp,
+  libusb1,
+  espeak ? null,
+  withEspeak ? false,
 }:
 
-stdenv.mkDerivation  rec {
-  pname = "rockbox-utility";
-  version = "1.4.1";
+stdenv.mkDerivation rec {
+  pname = "rbutil";
+  version = "1.5.1";
 
-  src = fetchurl {
-    url = "https://download.rockbox.org/rbutil/source/RockboxUtility-v${version}-src.tar.bz2";
-    hash = "sha256-PhlJ+fNY4/Qjoc72zV9WO+kNqF5bZQuwOh4EpAJwqX4=";
+  src = fetchgit {
+    url = "git://git.rockbox.org/rockbox.git";
+    rev = "rbutil_${version}";
+    hash = "sha256-n5UnqBhRJ8R4AV2PLNTHuTIRuyDSqV1CjkQGlOHPIUY=";
   };
 
   nativeBuildInputs = [
-    makeWrapper
+    cmake
     pkg-config
-    qt5.qmake
+    makeWrapper
     qt5.wrapQtAppsHook
   ];
 
   buildInputs = [
     cryptopp
     libusb1
     qt5.qtbase
+    qt5.qtmultimedia
     qt5.qttools
-  ]
-  ++ lib.optional withEspeak espeak;
+  ] ++ lib.optional withEspeak espeak;
 
-  postPatch = ''
-    sed -i rbutil/rbutilqt/rbutilqt.pro \
-        -e '/^lrelease.commands =/ s|$$\[QT_INSTALL_BINS\]/lrelease -silent|${lib.getDev qt5.qttools}/bin/lrelease|'
-  '';
+  cmakeFlags = [ "-Wno-dev" ];
 
-  preConfigure = ''
-    cd rbutil/rbutilqt
-    lrelease rbutilqt.pro
-  '';
-
-  # Workaround build failure on -fno-common toolchains like upstream
-  # gcc-10. Otherwise build fails as:
-  #   ld: libmkimxboot.a(elf.c.o):utils/imxtools/sbtools/misc.h:43: multiple definition of `g_nr_keys';
-  #     libmkimxboot.a(mkimxboot.c.o):utils/imxtools/sbtools/misc.h:43: first defined here
-  # TODO: try to remove with 1.5.1 update.
-  env.NIX_CFLAGS_COMPILE = "-fcommon";
+  preConfigure = "cd utils";
 
   installPhase = ''
-    runHook preInstall
+    install -Dm 755 rbutilqt/RockboxUtility -t $out/bin/
```

```suggestion
    installBin rbutilqt/RockboxUtility
```

And put `installShellFiles` in `nativeBuildInputs`

---

> ```suggestion
>     installBin rbutilqt/RockboxUtility
> ```
> 
> And put `installShellFiles` in `nativeBuildInputs`

Ok! What's preInstall for?

---

To run extra commands before the regular installation phase.

The `runHook preInstall` is here because of composability.